pkg/ha/state/state.go

```diff
@@ -56,20 +56,14 @@ func NewLease(c client.LeaseClient, cluster, potentialLeader string, minT, maxT,
 	if err != nil {
 		return nil, fmt.Errorf("could not create new lease: %#v", err)
 	}
-	var maxTimeInstance string
-	if maxT.After(dbState.LeaseUntil) {
-		maxTimeInstance = potentialLeader
-	} else {
-		maxTimeInstance = dbState.Leader
-	}
 
 	return &Lease{
 		cluster:               dbState.Cluster,
 		leader:                dbState.Leader,
 		leaseStart:            dbState.LeaseStart,
 		leaseUntil:            dbState.LeaseUntil,
 		maxTimeSeen:           maxT,
-		maxTimeInstance:       maxTimeInstance,
+		maxTimeInstance:       potentialLeader,
```

Comment:
I like this new name, For more clarity. :)

Reply:
May be should update the struct field from `maxTimeInstance` to `potentialLeader`